searx/plugins/self_info.py

```diff
@@ -25,20 +29,42 @@
 p = re.compile('.*user[ -]agent.*', re.IGNORECASE)
 
 
+# Returns a string with all the information retrieved from ip-api.com's API
+def get_ip(request):
+    x_forwarded_for = request.headers.getlist("X-Forwarded-For")
+
+    if x_forwarded_for:
+        ip = x_forwarded_for[0]
+    else:
+        ip = request.remote_addr
+
+    if ip == '127.0.0.1':
+        ip = ''
+
+    # Find the outgoing proxies settings from the user configuration.
+    outgoing_proxies = settings['outgoing'].get('proxies', None)
+
+    # Initiate a GET request and set the outgoing proxies, if any were set in
+    # settings.yml.
+    ip_info = get('http://ip-api.com/json/' + ip, proxies=outgoing_proxies).json()
```

Comment:
Seeing
* http://dev.maxmind.com/geoip/legacy/geolite/  (the new format has far less information)
* https://github.com/appliedsec/pygeoip/blob/master/docs/getting-started.rst

I would say that searx can handle that internally ? 
It would make sure that the IP is not spread. But when to download the database, when to update ?

Otherwise there is https://ipinfo.io/ (a comparaison tool : https://www.iplocation.net/ hoping it is fair).